docs/proposals/image-updater-first-class-support.md

```diff
@@ -0,0 +1,324 @@
+- Provides Image Updater functionality out of the box for all Argo CD installations even more natively
+- Image updater would receive native awareness of all applications in control-plane and non-control plane namespaces
+
+Drawbacks:
+
+- Loss of separation of concerns between application resource management and image updation that currently exists (it can probably still be achieved within the application-controller but the code may need to be structured more carefully)
+- Adds additional complexity to the application-controller that would now have to interact with image-registries & perform git write-backs as well (alternatively, git write-back functionality may potentially need to be extracted out and added to the repo-server instead)
+- Might impact maintenance/reliability of application-controller code
+- Separate controller would be consistent with how projects like appset and notifications controller have been merged into argo-cd in the past
+
+## Future enhancements
+
+At present there are some initiatives that are being considered which could enhance the usability and adoption of Image Updater with users in the future, some of which have been mentioned in this proposal already:
+
+1. Support for opening pull requests to source repositories and honoring rollbacks - At present Image Updater directly commits manifest updates to the target branch when git write-back is configured. However, if status information about the last conducted update and version updated to is available, Image Updater could use that information to potentially support creating pull requests (and not spam pull requests) and also not continuously try to upgrade to a new image version if an application has been rolled back. This is a feature that has been requested by users of the Image Updater. There are a number of use cases that would benefit from Pull request support in Image Updater, such as ones needing all changes to go through PRs as opposed to direct commits, requirments to run CI checks on PRs before merging image updates, rejecting PRs that try to pull in faulty images etc, among others. There are already some ideas for how this could be implemented. This would make for a great value-add as a future enhancement to the Image Updater's functionality.
+
+2. Change Image updater operational model to become webhook aware - At present the updater polls the configured image registries periodically to detect new available versions of images. In the future Image Updater would need to be able to support webhooks so that it can move to a pub/sub model and updates can be triggered as soon as a qualifying image version is found.
+
+3. Supporting container verification - As described earlier, this could be a useful feature to reduce security risks introduced into Argo CD while providing all the benefits of having automatic image updates available. 
```

Comment:
Suggestion: add better support for apps in mono-repo which is currently blocked by https://github.com/argoproj-labs/argocd-image-updater/issues/475#issuecomment-1557680629
I believe mono-repo is a common use case with ArgoCD and this support is crucial for adoption similar to other controllers in ArgoCD.